### ISSUE TITLE
Advanced contacts manager

### ISSUE THREAD

**Issue by florianch** [owncloud/contacts#743]
Hello,

I would like to create an advanced contact manager because the default one is too simple.

What I need :
All owncloud users are contacts
All owncloud groups of users are contacts groups
If you are in a group you see all contacts in this group
A contact can be public (see by all users) but he can hide (or show) phone number or email
A contact can be in many groups
The link beetween group and contact is a fonction ex: "florian" is "coordinator" of "local group" "florian" is "manager" of "ngo"
A contact have many skills (drawing, managing, it...) and many interests (ecology, economy, human rights...)
A group have many interest
Contacts can be linked for see each other if they aren't in same group
A plain text search in all fields

Do you have some tips, some advices, some code before I begin? Do you think it's better to edit your Contacts app or create a new one?

Thanks

**Comment by florianch** [comment]
Thanks for your responses.

I am the head of an NGO. With other NGOs we are creating a collective in our city. The group have currently a list of 70 NGOs, each one has one or more contacts. We see that the central element is the group, not the contact. Often people are very active, for example, the same person is president of the NGO X, secretary of the NGO Y and accounting of the NGO Z.
I tested various CRM and groupware but I did not find anything suitable, they are business oriented, that's why I need to develop a new application.

I do not know if using LDAP is better, I already used elsewhere and the group management is not flexible enough.

As you say, the Contacts app is not a user management but VCard addressbook, I do not know if I want to do is too complicated compared to the development of a small application totally separate.

**Comment by DeepDiver1975** [comment]
Generally speaking I'm not a fan of re-using owncloud user groups in any other context and wnhace it with additional relations and properties.

With respect to the current contacts app implementation the properies which can be assigned to an existing contact should be quite flexible (a VCard itself is just a list of key-value-pairs).

But I have no idea if properties can be assigned to the addressbooks - which would then the the equivalent of a group - with the restriction that one contacts can only be part of one group.

Maybe it would be worth thinking about an extended groups mechanism which would allow to add contacts to multiple groups.

**Comment by dagbdagb** [comment]
I could not find anything in the vcard 4.0 RFC about members of a group only belonging to a single group vcard.  See: https://tools.ietf.org/html/rfc6350#section-6.1.4

No idea how contemporary vcard implementations handle KIND="group".

Having a list of 'tags' per individual vcard would be better, but I can' see anything like that in the RFC. Adding individuals to a group vcard could still be implemented in the OC GUI by way of exposing a 'member' list per contact (checkboxes), and listing only values corresponding to existing group cards.

**Comment by babelouest** [comment]
this ticket and other (like #779 ) make me think that the contact editor in owncloud should be able to edit all possible fields, even the `X-` properties. We often are asked why this field or the other one are not handled in oc. Like this we could be able to edit all type of fields and even create new ones.

@florianch , in you case, would using a `X-` property like `X-NGO-GROUP` be useful to solve your problem ? This property would be specific to your needs and not misinterpreted by other vcard editors.

**Comment by jancborchardt** [comment]
Yeah, probably it should be possible. Normally these fields will not be present anyway – but if you have them they should be editable. Adding them can be possible by having the ability to add a custom field.
